File: src/main/java/com/sudosoo/takeiteasy/service/NoticeService.java
package com.sudosoo.takeiteasy.service;

import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;

public interface NoticeService {
    SseEmitter subscribe(String username, String lastEventId);
    void send(String receiver, String content);
    void kafkaSend(ConsumerRecord<String, String> record);

}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/service/NoticeService.kt
package com.sudosoo.takeItEasy.service

import org.apache.kafka.clients.consumer.ConsumerRecord

interface NoticeService {
    fun subscribe(username: String, lastEventId: String): SseEmitter
    fun send(receiver: String, content: String)
    fun kafkaSend(record: ConsumerRecord<String?, Any?>)
}
